internal/controller/generator/generator.go

```diff
@@ -2,16 +2,27 @@ package generator
 
 import (
 	"fmt"
-	"github.com/superhacker2002/parentheses/internal/parentheses"
 	"net/http"
 	"strconv"
 )
 
-func SetRoutes() {
-	http.HandleFunc("/generate", generateHandler)
+type parenthesesGenerator interface {
+	GenerateRandom(uint) string
 }
 
-func generateHandler(w http.ResponseWriter, r *http.Request) {
+type generator struct {
```

Comment:
not sure that it should have name like generator. It's just your http handler structure, so if you'll have more then one route and some of them would be not related to generate. I'll suggest to use some common name related to http and it's handling. Handler for example :)